src/python/bcc/__init__.py

```diff
@@ -218,12 +218,12 @@ def load_funcs(self, prog_type=KPROBE):
     def load_func(self, func_name, prog_type):
         if func_name in self.funcs:
             return self.funcs[func_name]
-
         if not lib.bpf_function_start(self.module, func_name.encode("ascii")):
             raise Exception("Unknown program %s" % func_name)
 
-        log_buf = ct.create_string_buffer(65536) if self.debug else None
-
+        prog_len = lib.bpf_function_size(self.module, func_name.encode("ascii"))
+        instruc_c = prog_len / 8 # sizeof(struct bpf_insn)
+        log_buf = ct.create_string_buffer(instruc_c * 128) if self.debug else None
```

Comment:
same here. if 65k was too small, make a logic to increase try load few times until kernel stops returning enospc